control_panel_api/tests/permissions/test_users3bucket_permissions.py

```diff
@@ -0,0 +1,165 @@
+from unittest.mock import MagicMock, patch
+
+from model_mommy import mommy
+from rest_framework.reverse import reverse
+from rest_framework.status import (
+    HTTP_200_OK,
+    HTTP_201_CREATED,
+    HTTP_204_NO_CONTENT,
+    HTTP_400_BAD_REQUEST,
+    HTTP_403_FORBIDDEN,
+)
+from rest_framework.test import APITestCase
+
+from control_panel_api.models import AccessToS3Bucket
+
+
+@patch('control_panel_api.aws.aws.client', MagicMock())
+class UserS3Buckets(APITestCase):
+    def setUp(self):
+        self.superuser = mommy.make(
+            'control_panel_api.User',
+            auth0_id='github|user_1',
+            is_superuser=True,
+        )
+        self.normal_user = mommy.make(
+            'control_panel_api.User',
+            username='alice normal',
+            auth0_id='github|user_2',
+            is_superuser=False,
+        )
+
+        self.user_1 = mommy.make("control_panel_api.User", is_superuser=False)
+
+        self.s3bucket_1 = mommy.make("control_panel_api.S3Bucket")
+        self.s3bucket_2 = mommy.make("control_panel_api.S3Bucket")
+
+        self.normal_user_admin = mommy.make(
+            "control_panel_api.UserS3Bucket",
+            user=self.normal_user,
+            s3bucket=self.s3bucket_1,
+            access_level=AccessToS3Bucket.READWRITE,
+            is_admin=True,
+        )
+        self.normal_user_other_bucket_non_admin = mommy.make(
+            "control_panel_api.UserS3Bucket",
+            user=self.normal_user,
+            s3bucket=self.s3bucket_2,
+            access_level=AccessToS3Bucket.READWRITE,
+            is_admin=False,
+        )
+        self.other_user_normal_user_has_admin = mommy.make(
```

Comment:
You may want to call this variable `other_user_admin` for consistency with `normal_user_admin`.

Reply:
Yeh it's named differently, because it's from the perspective of the normal_user. This one an other user owns it, but the bucket related the normal user has admin access.